### DISCUSSION TITLE
Is it possible to use Prisma to pull data from two tables at once?

### DISCUSSION THREAD

**Discussion by jonthewayne** [prisma/prisma#4136] [Q&A] [answered]
Say I have a `Photo` model and a `Video` model but I want to show users a view of their library made up of both photos and videos, and I want to show them all by most recently added. How can I use Prisma to grab from both models and sort the results correctly, do pagination on them, etc?

**Comment by ryands17** [comment, marked as answer]
Hey @jonthewayne 👋 
As the tables are not related, the best way to go would be a [raw query](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/raw-database-access).

Suppose this is my schema:

```prisma
model Photo {
  id        Int      @id @default(autoincrement())
  name      String
  url       String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model Video {
  id        Int      @id @default(autoincrement())
  name      String
  url       String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```

I could get the sorted results with pagination as follows:

```ts
import type { Photo } from '@prisma/client'

const feed = await prisma.$queryRaw<Photo[]>`
    select * from "Photo"
      union
    select * from "Video"
      order by "createdAt"
      limit 10
`
```

Here you can add as many conditions as you want and you will get the appropriate type checking by passing the type in `queryRaw`.

The second approach is not recommended as that could give you unexpected results and the order might not be correct.

Not recommended

```ts
const photos = await prisma.photo.findMany({
    orderBy: { createdAt: 'asc' },
    take: 5,
  })
  const videos = await prisma.photo.findMany({
    orderBy: { createdAt: 'asc' },
    take: 5,
  })

  // perform sorting in code here (might possibly be ordered wrong)
```

Let me know if that helps.

**Reply by jonthewayne** [reply]
Hey @ryands17 :)

I think you are right about approach #1 being the best one. 

If I wanted to implement pagination, would I basically be re-implementing what Prisma gives us automatically?

I was reading in the Prisma docs that you can type the results of a raw sql, but how do I use a specific type depending on which table the results come from?

Also, wouldn't this be a really nice feature to have with Prisma? :)

**Reply by ryands17** [reply]
@jonthewayne I updated my answer with the types :)

As for pagination, unfortunately as it's a raw query so yes you would be implementing what Prisma provides automatically.

> I was reading in the Prisma docs that you can type the results of a raw sql, but how do I use a specific type depending on which table the results come from?

When using a `union`, the columns taken will be the same, so it doesn't matter where the type comes from, so you can use the type of any table. As for a feature request, I would suggest you to create one [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
